### PR TITLE
Fixed #33866 -- Added pathlib.Path support to GDALRaster constructor. 

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: tests/gis_tests/gdal_tests/test_raster.py [hunk 1]
```diff
@@ -25,6 +26,11 @@ def setUp(self):
         )
         self.rs = GDALRaster(self.rs_path)
 
+    def test_gdalraster_input_as_path(self):
+        rs_path = Path(__file__).parent.joinpath("data", "rasters", "raster.tif")
```
Comment (felixxm) [inline tests/gis_tests/gdal_tests/test_raster.py:30]:
```suggestion
        rs_path = Path(__file__).parent / "data" /  "rasters" / "raster.tif"
```

File: tests/gis_tests/gdal_tests/test_raster.py [hunk 2]
```diff
@@ -25,6 +26,11 @@ def setUp(self):
...
+        GDALRaster(rs_path)
```
Comment (felixxm) [inline tests/gis_tests/gdal_tests/test_raster.py:31]:
```suggestion
        self.rs = GDALRaster(rs_path)
```

**Comment by claudep** [comment]
Sorry @felixxm, I pushed without my latest changes :-(

**Review by felixxm — COMMENTED** [review]
@claudep Thanks :+1: I pushed docs changes.

**Comment by claudep** [comment]
Nice!

**Pull request merged**